test/Vehicle/SendMavCommandWithSignallingTest.cc

```diff
@@ -38,6 +38,18 @@ void SendMavCommandWithSignallingTest::_testCaseWorker(TestCase_t& testCase)
 
     QCOMPARE(spyResult.wait(10000), true);
     QList<QVariant> arguments = spyResult.takeFirst();
+
+    // Gimbal controler requests MAVLINK_MSG_ID_GIMBAL_MANAGER_INFORMATION on vehicle connection,
+    // and that messes with this test, as it receives response to that command instead. So if we 
+    // are taking the response to that MAV_CMD_REQUEST_MESSAGE, we discard it and take the next
+    // Also, the camera manager requests MAVLINK_MSG_ID_CAMERA_INFORMATION on vehicle connection,
+    // so we need to ignore that as well.
+    while (arguments.at(2).toInt() == MAV_CMD_REQUEST_MESSAGE || arguments.at(2).toInt() == MAV_CMD_REQUEST_CAMERA_INFORMATION) {
```

Comment:
This is kinda nasty and fragile. There is already the concept of creating a connect to a mock link vehicle which does not do the initial vehicle connect sequence. This is specifically for unit test problems like this. Would it be possible to hook in the gimbal request stuff into the vehicle initial connect sequence state machine so that a it only happens if that was run?

Reply:
I can try that but I will have to do a bit more reading to understand these tests. It's all a mystery to me.

Reply:
Don't worry about it. I'll clean this up after it's merged in. Will be easier for me to deal with it than have you figure out something new to you.

Reply:
Same thing the the request message stuff. I can switch that over as well.